internal/charmstore/resource.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package charmstore // import "gopkg.in/juju/charmstore.v5-unstable/internal/charmstore"
+import (
+	"sort"
+
+	"gopkg.in/juju/charm.v6-unstable"
+	"gopkg.in/mgo.v2/bson"
+
+	"gopkg.in/juju/charmstore.v5-unstable/internal/mongodoc"
+)
+
+// newResourceQuery formats the provided details into a mongo query for
```

Comment:
// newResourceQuery returns a mongo query doc that will retrieve the
// given named resource and revision associated with the given
// charm URL.

?